extensions/resource-deployment/src/services/tools/toolBase.ts

```diff
@@ -19,47 +20,170 @@ export abstract class ToolBase implements ITool {
+			await this.checkAndUpdateVersion();
+			updateDisplayTableData.call(thisObject, this);
+		} catch (error) {
+			const errorMessage = this._platformService.getErrorMessage(error);
+			this._statusDescription = localize('deployCluster.InstallError', "Error installing tool '{0}'.{1}Error: {2}{1}See output channel '{3}' for more details", this.displayName, EOL, errorMessage, this.outputChannelName);
+			this._status = ToolStatus.Error;
+			updateDisplayTableData.call(thisObject, this);
+			throw error;
+		}
+	}
+
+	protected async installCore() {
+		const installationCommands: Command[] = this.installationCommands;
+		if (!installationCommands || installationCommands.length === 0) {
+			throw new Error(`Cannot install tool:${this.displayName}::${this.description} as installation commands are unknown`);
+		}
+		for (let i: number = 0; i < installationCommands.length; i++) {
+			await this._platformService.runCommand(installationCommands[i].command,
+				{
+					workingDirectory: installationCommands[i].workingDirectory || this.downloadPath,
+					additionalEnvironmentVariables: installationCommands[i].additionalEnvironmentVariables
+				},
+				installationCommands[i].sudo,
+				installationCommands[i].comment,
+				installationCommands[i].ignoreError
+			);
+		}
+	}
+
+	protected async addInstallationSearchPathsToSystemPath(): Promise<void> {
+		const installationPath = await this.getInstallationPath();
+		const searchPaths = [installationPath, ...this.installationSearchPaths].filter(path => !!path);
+		this.logToOutputChannel(`Search Paths for tool '${this.displayName}': ${JSON.stringify(searchPaths, undefined, '\t')}`);
+		searchPaths.forEach(installationSearchPath => {
+			if (installationSearchPath) {
+				if (process.env.PATH) {
+					if (!`${delimiter}${process.env.PATH}${delimiter}`.includes(`${delimiter}${installationSearchPath}${delimiter}`)) {
+						process.env.PATH += `${delimiter}${installationSearchPath}`;
+						console.log(`Appending to Path -> ${delimiter}${installationSearchPath}`);
+					}
+				} else {
+					process.env.PATH = installationSearchPath;
+					console.log(`Appending to Path -> '${delimiter}${installationSearchPath}':${delimiter}${installationSearchPath}`);
+				}
 			}
-		}).catch((error) => {
-			const errorMessage = typeof error === 'string' ? error :
-				typeof error.message === 'string' ? error.message : '';
-			this._statusDescription = localize('deployCluster.GetToolVersionError', "Error retrieving version information.{0}Error: {1}{0}stdout: {2} ", EOL, errorMessage, this._versionOutput);
 		});
 	}
+	public async loadInformation(): Promise<void> {
+		if (this.status === ToolStatus.Installed) {
+			return Promise.resolve();
+		}
+		if (this.status === ToolStatus.NotInstalled) {
+			this._statusDescription = undefined;
+			this._version = undefined;
+			this._versionOutput = undefined;
+			this._osType = this._platformService.osType();
+			await this.addInstallationSearchPathsToSystemPath();
```

Comment:
Why not just set this on creation? This should never change I'd imagine

Reply:
fixed

Reply:
Yes moved the initialization out to constructor.